tools/bumper/cnao_repo_commands.go

```diff
@@ -212,6 +215,138 @@ func (cnaoRepoOps *gitCnaoRepo) collectModifiedToTreeList(allowedList []string)
 	return entries, nil
 }
 
+// getBumpBranch creates a new bump branch.
+// Since we don't want to use already open branch
+// that may hold outdated/needs rebasing changes
+// we concat a random string to the end of the branch name
+func (cnaoRepoOps *gitCnaoRepo) getBumpBranch(prTitle string) (*github.Reference, string, error) {
+	branchBaseName := strings.Replace(strings.ToLower(prTitle), " ", "_", -1)
+	branchOpenMaxTries := 10
+	for branchIdx := 0; branchIdx < branchOpenMaxTries; branchIdx++ {
+		branchNameWithRandSuffix := fmt.Sprintf("%s_%s", branchBaseName, rand.String(5))
+		branchRef, _, err := cnaoRepoOps.githubInterface.GetBranchRef(cnaoRepoOps.getCnaoRepoOwnerFromUrl(), cnaoRepoOps.getCnaoRepoNameFromUrl(), "refs/heads/"+branchNameWithRandSuffix)
+		if err != nil {
+			if cnaoRepoOps.branchNotFound(branchRef, err) {
+				return cnaoRepoOps.createNewGithubBranch(branchNameWithRandSuffix)
+			} else {
+				return nil, "", errors.Wrap(err, "Failed to check if branch is already open")
+			}
+		}
+
+		logger.Printf("branch %s already open: modifing branch name", branchNameWithRandSuffix)
+	}
+
+	return nil, "", fmt.Errorf("Failed to create a new branch from base name %s after %d tries", branchBaseName, branchOpenMaxTries)
+}
+
+func (cnaoRepoOps *gitCnaoRepo) branchNotFound(branchRef *github.Reference, err error) bool {
+	if branchRef == nil && strings.Contains(err.Error(), "404 Not Found") {
```

Comment:
We cannot compare with proper error from githubInterface instead of string at Errors ? 

https://blog.golang.org/go1.13-errors, look for `Examining errors with Is and As`

Reply:
I found the proper way to check if the request got `not found` : https://github.com/google/go-github/blob/76c3c3d7c6e78e8c91e77d2e2578c4e0a7cf96ea/github/git_refs_test.go#L73
fixed